3. Java Servlet Web App Tutorial/1-Authentication/sign-in-auth-context/README.md

```diff
@@ -0,0 +1,178 @@
+1. Select **Save** to save your changes.
+1. In the app's registration screen, click on the **Certificates & secrets** blade in the left to open the page where we can generate secrets and upload certificates.
+1. In the **Client secrets** section, click on **New client secret**:
+    - Type a key description (for instance `app secret`),
+    - Select one of the available key durations (**In 1 year**, **In 2 years**, or **Never Expires**) as per your security concerns.
+    - The generated key value will be displayed when you click the **Add** button. Copy the generated value for use in the steps later.
+    - You'll need this key later in your code's configuration files. This key value will not be displayed again, and is not retrievable by any other means, so make sure to note it from the Azure portal before navigating to any other screen or blade.
+
+
+### Configure the web app (java-servlet-webapp-authentication) to use your app registration
+
+Open the project in your IDE to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `./src/main/resources/authentication.properties` file
+2. Find the string `{enter-your-tenant-id-here}`. Replace the existing value with you tenant id
+3. Find the string `{enter-your-client-id-here}` and replace the existing value with the application ID (clientId) of the `java-servlet-webapp-authentication` application copied from the Azure portal.
+4. Find the string `{enter-your-client-secret-here}` and replace the existing value with the key you saved during the creation of the `java-servlet-webapp-authentication` app, in the Azure portal.
+
+</details>
+
+### 2. Setup the MFA Conditional Access Context
+
+1. In the Azure Portal, go to the Microsoft Entra Conditional Access section.
+
+![CA](./ReadmeFiles/CA.png)
+
+Note: this is a different Azure Portal section than Microsoft Entra ID
+
+2. Go to the "Authentication Contexts" page, and create a new context. Give it a name (e.g. "MFA") and select and id from the list, for example "c1". Ensure the "Publish To Apps" checkbox is checked. Make a note of the ID.
+3. Now go to Policies and create a new Conditional Access policy. 
+    - Select "All Users" or a subset of users
+    - Select "Target Resources" and set it to "Authentication Context". Then add the context you just created.
+    - You can skip the "Conditions" part of the flow. This policy will apply to all calls.
+    - In Grant, check the "Require Multi-Factor Authentication" policy.
+    - You can skip the "Session" 
+
+![CA-MFA](./ReadmeFiles/CA-MFA.png)
+
+## Running The Sample
+
+#### Build .war File Using Maven
+
+1. Navigate to the directory containing the pom.xml file for this sample (the same directory as this README), and run the following Maven command:
+    ```
+    mvn clean package
+    ```
+1. This should generate a `.war` file which can be run on a variety of application servers
```

Comment:
```suggestion
1. If the build is successful, it will generate a `.war` file in the `target` folder which can be run on a variety of application servers.
```